test/cpp/security/tls_certificate_verifier_test.cc

```diff
@@ -161,6 +161,85 @@ TEST(TlsCertificateVerifierTest,
   EXPECT_EQ(sync_status.error_message(), "Hostname Verification Check failed.");
 }
 
+TEST(TlsCertificateVerifierTest,
+     NoOpCertificateVerifierSucceedsWithVerifiedRootCertSubject) {
+  grpc_tls_custom_verification_check_request request;
+  memset(&request, 0, sizeof(request));
+  char* expected_subject = const_cast<char*>(
+      "CN=testca,O=Internet Widgits Pty Ltd,ST=Some-State,C=AU");
+  request.peer_info.verified_root_cert_subject = expected_subject;
+  auto verifier = std::make_shared<NoOpCertificateVerifier>();
+  TlsCustomVerificationCheckRequest cpp_request(&request);
+  EXPECT_EQ(cpp_request.verified_root_cert_subject(), expected_subject);
+  grpc::Status sync_status;
+  verifier->Verify(&cpp_request, nullptr, &sync_status);
```

Comment:
Should we add an expectation on the return value of the `verifier->Verify` call?

Similarly throughout.

Reply:
The return on this indicates whether it executed synchronously or not - actually checking the status of the verification is done in the line below checking `sync_status`.
Given this is a sync test, I've gone on and added a check that it returns `True`.